File: QytetetJava/src/modeloqytetet/Dado.java

package modeloqytetet;

/**
 *
 * @author albertorodriguez
 */
class Dado {
    
    
    //AL SER UNA CLASE SINGLETON, COMO DEVOLVER EL OBJETO.
    
    private static final Dado instance = new Dado();
    
    //Constructor privado
    private Dado(){}
    
    public static Dado getInstance(){
        return instance;
    }
    
    int tirar(){
      throw new UnsupportedOperationException("Sin Implementar");
    }
    
    
    
}

File: Practicas/Examen 1/QytetetRuby/pkg/QytetetRuby-0.0.1/lib/dado.rb
#encoding: utf-8

require "singleton"

module ModeloQytetet
  
  
  class Dado
    include Singleton
    #PARA ACCEDER A LA ISNTANCIA DE LA CLASE DADO EN OTRA CLASE -> dado = Dado.instance
    
    def tirar
    end
    
    def initialize
    end
    
  end
end


